avogadro/qtplugins/svg/svg.cpp

```diff
@@ -0,0 +1,259 @@
+    m_idToindex[m.id] = i;
+    m.pos_image = posToImage(m.pos_view, m.r);
+    ++i;
+  }
+}
+
+// paint all bondings only once
+void SVG::paintBonds(QPainter& painter, const Mol& mol, unsigned int i,
+                     const NeighborListType& bonds)
+{
+  painter.setPen(QPen(QColor(125, 125, 125), PEN_WIDTH_BOND));
+  for (auto it = bonds.begin(); it != bonds.end(); ++it) {
+    unsigned int j = it->atom2().index();
+    if (m_idToindex[j] == i) {
+      j = it->atom1().index();
+    }
+    if (m_idToindex[j] <= i) {
+      continue;
+    }
+    // calculate the inicial and final position, considering the mol middle
+    // and the middle of the pen width
+    auto mol_to = m_mols[m_idToindex[j]].pos_image;
+    Eigen::Vector2f from(mol.pos_image[0] - (PEN_WIDTH_BOND / 2.0f),
+                         mol.pos_image[1] - (PEN_WIDTH_BOND / 2.0f));
+    Eigen::Vector2f to(mol_to[0] - (PEN_WIDTH_BOND / 2.0f),
+                       mol_to[1] - (PEN_WIDTH_BOND / 2.0f));
+
+    float L = std::sqrt((from[0] - to[0]) * (from[0] - to[0]) +
+                        (from[1] - to[1]) * (from[1] - to[1]));
+    float offsetX = (to[1] - from[1]) / L;
+    float offsetY = (from[0] - to[0]) / L;
+    unsigned int order = int(it->order());
+    // for each bound offset it following the orthogonal direction
+    for (unsigned int o = 0; o < order; ++o) {
+      // if there is only one bond, don't displace
+      float x = 0, y = 0;
+      if (order > 1) {
+        x = (float(o) - (order / 2.0f)) * OFF_SET_PARALEL * offsetX;
+        y = (float(o) - (order / 2.0f)) * OFF_SET_PARALEL * offsetY;
+      }
+      QLineF line(from[0] + x, from[1] + y, to[0] + x, to[1] + y);
+      painter.drawLine(line);
+    }
+  }
+}
+
+void SVG::paintMol(QPainter& painter, const Mol& mol)
+{
+  Vector3ub color_mol = m_molecule->color(mol.id);
+  QColor color(color_mol[0], color_mol[1], color_mol[2]);
+  painter.setPen(QPen(QColor(0, 0, 0), PEN_WIDTH_MOL));
+  painter.setBrush(color);
+  painter.drawEllipse(mol.pos_image[0] - mol.pos_image[2] / 2.0f,
+                      mol.pos_image[1] - mol.pos_image[2] / 2.0f,
+                      mol.pos_image[2], mol.pos_image[2]);
+}
+
+void SVG::paintSVG(QPainter& painter)
+{
+  const Vector4ub background = m_scene->backgroundColor();
+  QColor backgroundColor(background[0], background[1], background[2],
+                         background[3]);
+  painter.fillRect(QRect(0, 0, m_width, m_height), backgroundColor);
+  getPositions();
+  for (unsigned int i = 0; i < m_mols.size(); ++i) {
+    auto mol = m_mols[i];
+    const NeighborListType bonds = m_molecule->bonds(mol.id);
+    if (frustrumCulling(mol)) {
+      paintBonds(painter, mol, i, bonds);
+      paintMol(painter, mol);
+    }
+  }
+}
+
+void SVG::render()
+{
+  if (defaultChecks())
+    return;
+  QString filename = QFileDialog::getSaveFileName(
+    qobject_cast<QWidget*>(parent()), tr("Save File"), QDir::homePath(),
+    tr("SVG (*.svg)"));
+  QFile file(filename);
```

Comment:
I don't know if it's only my Mac (likely) but I can save a filename without an SVG extension(!)
```
if (!filename.endsWith(".svg", Qt::CaseInsensitive))
   filename.append(".svg");
```